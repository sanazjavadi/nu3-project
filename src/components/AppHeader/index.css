.App-Header {
  background-color: #282c34;
  min-height: 60px;
  max-width: 1200px;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
  margin: 0 auto;
}
