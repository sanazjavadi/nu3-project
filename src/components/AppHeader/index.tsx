import React, { FunctionComponent } from 'react';
import './index.css';

export interface AppHeaderProps {
  title: string;
}

export const AppHeader: FunctionComponent<AppHeaderProps> = ({ title }): JSX.Element =>  
<>
  <header className="App-Header">
    <h1>{title}</h1>
  </header>
</>

export default AppHeader;
