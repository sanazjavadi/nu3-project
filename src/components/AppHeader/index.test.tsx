import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import AppHeader from './';

test('AppHeader: Renders', () => {
  render(<AppHeader title="Demo Header" />);
  const linkElement = screen.getByText(/demo header/i);
  expect(linkElement).toBeInTheDocument();
});

test('AppHeader: Snapshot', () => {
  const component = renderer.create(<AppHeader title="Demo Header" />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
