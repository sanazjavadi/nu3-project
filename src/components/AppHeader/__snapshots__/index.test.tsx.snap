// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppHeader: Snapshot 1`] = `
<header
  className="App-Header"
>
  <h1>
    Demo Header
  </h1>
</header>
`;
